File: docs-examples/example-java/src/test/java/io/micronaut/ignite/docs/config/IgniteConfigurationFactoryInterceptor.java
package io.micronaut.ignite.docs.config;

import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import org.apache.ignite.configuration.IgniteConfiguration;

import javax.inject.Singleton;

@Singleton
public class IgniteConfigurationFactoryInterceptor implements BeanCreatedEventListener<IgniteConfiguration> {
    @Override
    public IgniteConfiguration onCreated(BeanCreatedEvent<IgniteConfiguration> event) {
        IgniteConfiguration configuration = event.getBean();
        configuration.setIgniteInstanceName("instance-a");
        return configuration;
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/ignite/docs/config/IgniteConfigurationFactoryInterceptor.kt
package io.micronaut.ignite.docs.config

import io.micronaut.context.event.BeanCreatedEvent
import io.micronaut.context.event.BeanCreatedEventListener
import org.apache.ignite.configuration.IgniteConfiguration

class IgniteConfigurationFactoryInterceptor : BeanCreatedEventListener<IgniteConfiguration> {
    override fun onCreated(event: BeanCreatedEvent<IgniteConfiguration>?): IgniteConfiguration {
        val configuration = event!!.bean
        configuration.igniteInstanceName = "instance-a"
        return configuration
    }
}
